card/mod/carrierwave/set/abstract/attachment/storage_type.rb

```diff
@@ -42,6 +47,11 @@
   @new_storage_type ||= storage_type_from_config
 end
 
+event :change_bucket_if_read_only, :initialize, on: :update, when: :cloud? do
```

Comment:
when possible, I like to do things like:
```
  event :change_bucket, when: read_only_cloud?...
```
that way, you can have another event that is `after: :change_bucket`, and neither happens unless the condition is met.